test_cluster_detection.py

```diff
@@ -0,0 +1,423 @@
+    return detections, y
+
+class Test_Model_Execution_Condition():
+    """
+    Tests execution condition for estimation worker at different amount of total and new data points
+    """
+
+    def test_under_min_total_threshold(self, cluster_model):
+        """
+        Total data under threshold should not run 
+        """
+        # Setup 
+        NUM_DATA_POINTS = MIN_TOTAL_POINTS_THRESHOLD - 1  # less than min threshold (100)
+        X, y = cluster_data([NUM_DATA_POINTS])
+
+        # Run
+        model = cluster_model
+        model.clear_all_data()
+        model_ran, detections_in_world = model.run(X, False)
+        
+        # Test
+        assert(model_ran == False)
+        assert(detections_in_world == None)
+    
+    def test_at_min_total_threshold(self, cluster_model):
+        """
+        Should run once once total threshold reached regardless of
+        current bucket size
+        """
+        # Setup
+        NUM_DATA_POINTS = MIN_TOTAL_POINTS_THRESHOLD - 1  # should not run the first time
+        NEW_DATA_POINTS = MIN_TOTAL_POINTS_THRESHOLD - 1 # under 10 new points 
+        
+        X, y = cluster_data([NUM_DATA_POINTS])
+        X_2, y_2 = cluster_data([NEW_DATA_POINTS])
+
+        # Run
+        model = cluster_model
+        model.clear_all_data()
+        model_ran, detections_in_world = model.run(X, False)
+        model_ran, detections_in_world = model.run(X_2, False)
+
+        # Test
+        assert(model_ran == True)
+        assert(detections_in_world != None)
+
+    def test_under_min_bucket_size(self, cluster_model):
+        """
+        New data under threshold should not run 
+        """
+        # Setup
+        NUM_DATA_POINTS = MIN_TOTAL_POINTS_THRESHOLD + 10  # should run the first time
+        NEW_DATA_POINTS = MIN_NEW_POINTS_THRESHOLD - 1 # under 10 new points 
+
+        X, y = cluster_data([NUM_DATA_POINTS])
+        X_2, y_2 = cluster_data([NEW_DATA_POINTS])
+
+        # Run
+        model = cluster_model
+        model.clear_all_data()
+        model_ran, detections_in_world = model.run(X, False)
+        model_ran, detections_in_world = model.run(X_2, False)
+
+        # Test
+        assert(model_ran == False)
+        assert(detections_in_world == None)
+
+    def test_good_data(self, cluster_model):
+        """
+        All conditions met should run
+        """
+        NUM_TRUE_CLUSTERS = MIN_TOTAL_POINTS_THRESHOLD + 1  # more than min total threshold should run 
+        X, y = cluster_data([NUM_TRUE_CLUSTERS])
+        model = cluster_model
+    
+        # Run
+        model = cluster_model
+        model.clear_all_data()
+        model_ran, detections_in_world = model.run(X, False)
+        
+        # Test
+        assert(model_ran == True)
+        assert(detections_in_world != None)
+    
+
+class Test_Correct_Number_Cluster_Outputs():
+    """
+    Tests if cluster estimation output matches number of input clusters
+    at different data scenarios. 
```

Comment:
For each of the 5 cluster tests, test with a single cluster.

Reply:
Can you clarify what you mean by this? Do you mean change each of the tests under the TestCorrectNumberClusterOutputs to only generate and check 1 cluster center instead of 5?

Reply:
For each test, also add a test for a single cluster.